src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -642,6 +642,35 @@ At this point of the lifecycle, you could choose to define your own swap impleme
 </script>
 ```
 
+#### Building blocks for custom swap functions
+
+<p><Since v="4.15.0" /></p>
+
+Astro offers utility functions that can be used to define a custom swap implementation.
+The `swapFunction` object of the `astro:transitions/client` module provides several reusable building blocks. The following example demonstrates how to combine these blocks to re-create Astro's built-in swap implementation.
```

Comment:
```suggestion
The `swapFunction` object of the `astro:transitions/client` module defines a `swap` implementation with five utility functions to describe specific actions related to page elements, attributes, and scripts during the swap transition. These can be used directly to define a custom swap implementation.

The following example uses these six functions to Astro's built-in `swap` implementation:
``` 
This might not be entirely accurate, but what about something in this style?